sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -630,4 +633,54 @@ private static <T> T callHandler(Supplier<T> callable, ExecutorService threadPoo
             }
         }
     }
+
+
+    public static Map<String, Object> adjustOdataPropertyMap(Map<String, Object> propertyMap) {
+        Map<String, Object> adjustedMap = new HashMap<>(propertyMap);
+
+        for (String key : propertyMap.keySet()) {
+            if (!key.endsWith("@odata.type")) {
+                Object convertedOdataObject = adjustedMap.computeIfPresent(key.concat("@odata.type"), (k, v) -> {
+                    String odataType = (String) v;
+                    if ("Edm.Binary".equals(odataType)) {
+                        if (!(adjustedMap.get(key) instanceof byte[])) {
+                            byte[] bytes = java.util.Base64.getDecoder().decode(((String) adjustedMap.get(key)).getBytes());
+                            return bytes;
+                        }
+                    } else if ("Edm.DateTime".equals(odataType)) {
```

Comment:
These string literals are in at least two places in the code now.

Can you make these `protected static final` and use them here and in the EntityDataModelType where they're also used?